### PR TITLE
add search/filter and middleware

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
[- ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/75203)
https://edu-redmine.sun-asterisk.vn/issues/74383?issue_count=19&issue_position=4&next_issue_id=74382&prev_issue_id=74406
## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/public/js/product/product.js [hunk 1]
```diff
@@ -13,3 +13,96 @@ if (btns) {
     });
   });
 }
+
+const getValueFromQueryString = (key) => {
+  const urlParams = new URL(window.location.href).searchParams;
+  if (urlParams.has(key)) {
+    return urlParams.get(key);
+  }
+  return urlParams.get(key);
```
Comment (huongnt-2545) [inline src/public/js/product/product.js]: đều cùng return 1 kết quả thì cần gì check điều kiện ở đây e hè

File: src/views/product/index.hbs [hunk 2]
```diff
@@ -2,21 +2,45 @@
...
+                        <input type="text" class="form-control" id="minRange" min="0" placeholder="Min Range">
+                    </div>
+                    <div class="col">
+                        <input type="text" class="form-control" id="maxRange" min="0" placeholder="Max Range">
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+    <button class="btn-clear-filter btn-primary-bg" id="clearFilter">Clear
```
Comment (huongnt-2545) [inline src/views/product/index.hbs]: I18n text

**Comment by ghost** [comment]
ready

**Pull request merged**